modules/tasks.go

```diff
@@ -443,6 +443,10 @@ func (t *Task) addToSchedule(overtime bool) {
 }
 
 func waitUntilNextScheduledTask() <-chan time.Time {
+	if sleepMode.IsSet() {
+		<-taskSchedulerSleepModeExitChannel
```

Comment:
We should return a read channel here, so that the task scheduler can shut down during sleep.